gms/gossiper.cc

```diff
@@ -1814,8 +1814,12 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
     _enabled = true;
     _nr_run = 0;
     _scheduled_gossip_task.arm(INTERVAL);
+    _background_msg = seastar::gate();
+    /* Ensure all shards have enabled gossip before starting the failure detector loop */
     co_await container().invoke_on_all([] (gms::gossiper& g) {
         g._enabled = true;
+    });
+    co_await container().invoke_on(0, [] (gms::gossiper& g) {
```

Comment:
With this patch 
Try this:
```
nodetool disablegossip
nodetool enablegossip
nodetool disablegossip 
```
You will see the node crashing.

The reason is that you changed to invoke on zero here. While the stop part is still on all shards.

With the following patch, it will work.

```
-        container().invoke_on_all([] (auto& g) {
+        container().invoke_on(0, [] (auto& g) {
             return std::move(g._failure_detector_loop_done);
         }).get();
```

However,  I think it is better to just call invoke on all everywhere. 

So the complete patch looks like:

```
--- a/gms/gossiper.cc
+++ b/gms/gossiper.cc
@@ -1823,8 +1823,14 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
     _enabled = true;
     _nr_run = 0;
     _scheduled_gossip_task.arm(INTERVAL);
+    if (!_background_msg.is_closed()) {
+        co_await _background_msg.close();
+    }
+    _background_msg = seastar::gate();
     co_await container().invoke_on_all([] (gms::gossiper& g) {
         g._enabled = true;
+    });
+    co_await container().invoke_on_all([] (gms::gossiper& g) {
         g._failure_detector_loop_done = g.failure_detector_loop();
     });
 }
```

Reply:
Done. Since the failure_detector_loop only execute on shard 0 I didn't expect that it should be executed on all shards but now it's clear, thanks